lib/Listener/BeforeGetTemplatesListener.php

```diff
@@ -20,14 +20,14 @@ public function __construct(
 	}
 
 	public function handle(Event $event): void {
-		if (!$event instanceof BeforeGetTemplatesEvent) {
+		if (!$event instanceof BeforeGetTemplatesEvent || !$event->shouldGetFields()) {
 			return;
 		}
 
```

Comment:
To make sure richdocuments still works with the old version (and have an easy to read conditions): 
```suggestion
		if (!$event instanceof BeforeGetTemplatesEvent) {
			return;
		}
		
		if (method_exists($event, 'shouldGetFields') && !$event->shouldGetFields()) {
			return;
		}
```